src/include/sof/audio/component.h

```diff
@@ -152,6 +152,26 @@ enum {
 /** \brief Retrieves subid (comp id) from the component device */
 #define trace_comp_get_subid(comp_p) ((comp_p)->ipc_config.id)
 
+#ifdef __ZEPHYR__
+/* class (driver) level (no device object) tracing */
+#define comp_cl_err(drv_p, __e, ...) LOG_ERR(__e, ##__VA_ARGS__)
```

Comment:
any reason for keeping these legacy macros in common code ?

Reply:
We need to proceed in steps. This first, very big step is switching the "backend" to the Zephyr logging system. Once it's passing some tests we can start some massive search/replace.

Reply:
Its already passing tests as we are using it in the Zephyr wrapper, this is just unnecessary churn. The aim of this PR should be to use the native Zephyr API and move the wrapper to xtos.

Reply:
We have a bit of Zephyr logging already but it's not tested at all.

Reply:
That's fine, the Zephyr parts are tested by Zephyr CI. This gets everything ready for the DMA backend which will use the same kernel interface and flow control on current HW, albeit with a different trace tool from Zephyr.

Reply:
Maybe I wasn't very clear sorry: if we merge this now, we lose all firmware logs in all our Zephyr test results.

Reply:
So in this case the xtos wrapper for LOG would be used until the Zephyr DMA backend is ready. The etrace parts should work.

Reply:
For zephyr logging, context is define per module/file, when we use zephyr api, we don't need to pass a context pointer to it. but for XTOS logging calls, we need to pass a context to it. That's why I still use the tr_xxx here, for zephyr, it will be replaced with LOG_XXX, and the context pointer is ignored, for XTOS, everything keeps unchanged.

> any reason for keeping these legacy macros in common code ?

Directly replace tr_xxx with zephyr LOG_XXX and provide wrapper for XTOS is not OK, because LOG_XXXs don't know how to handle the context pointer. Or I get it wrong here?